// blizzapi/dist/constants/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const localization = __importStar(require("./localization"));
const api = __importStar(require("./api"));
const oauth_1 = __importDefault(require("./oauth"));
const CONSTANTS = {
    REGIONS: localization.regions,
    LOCALES: localization.locales,
    DEFAULT_LOCALES: localization.defaultLocales,
    SC2_REALMS: localization.sc2Realms,
    REGION_API_HOSTS: api.regionHosts,
    OAUTH_AUTHORIZE_URIS: oauth_1.default.authorizeUris,
    OAUTH_TOKEN_URIS: oauth_1.default.tokenUris,
    OAUTH_CHECK_TOKEN_URIS: oauth_1.default.checkTokenUris,
};
exports.default = CONSTANTS;
